Sources/JXKit/JXConvertible.swift

```diff
@@ -121,12 +96,160 @@ extension Array: JXConvertible where Element: JXConvertible {
     }
 }
 
-extension JXValue {
-    /// Sets a `JXConvertible` in this value object.
-    /// - Parameters:
-    ///   - key: The key to set.
-    ///   - object: The `JXConvertible` to convert.
-    public func set<T: JXConvertible>(_ key: String, convertible object: T) throws {
-        try setProperty(key, object.toJX(in: self.context))
+extension Dictionary: JXConvertible where Key == String, Value: JXConvertible {
+    public static func fromJX(_ value: JXValue) throws -> Dictionary<Key, Value> {
+        guard value.isObject else {
+            throw JXErrors.valueNotObject
+        }
+        let jxDictionary = try value.dictionary
+        return try jxDictionary.reduce(into: [:]) { result, entry in
+            result[entry.key] = try Value.fromJX(entry.value)
+        }
+    }
+
+    public func toJX(in context: JXContext) throws -> JXValue {
+        let jxDictionary = try self.reduce(into: [:]) { result, entry in
+            result[entry.key] = try entry.value.toJX(in: context)
+        }
+        return try context.object(fromDictionary: jxDictionary)
+    }
+}
+
+extension RawRepresentable where RawValue: JXConvertible {
```

Comment:
Moved all this from Jack. I changed the numeric ones so that we use a single code path through context.number and jxvalue.xxx